### PR TITLE
New additions

### PR DESCRIPTION
**Pull request by sourabhdalvi**
- [x] Add Auto-generated structs
- [x] Add Serialization 
- [x] Add De-serialization
- [ ] Add Test

### CODE REVIEW

File: src/descriptors/portfolio_structs.json [hunk 1]
```diff
@@ -0,0 +1,292 @@
+{
```
Comment (jd-lara) [inline src/descriptors/portfolio_structs.json:1]: why we need autogeneration for this application?. This is more complicated that keeping simple types.

**Comment by jd-lara** [comment]
> * [x]  Add Auto-generated structs
> * [x]  Add Serialization
> * [ ]  Add De-serialization
> * [ ]  Add Test

We discussed supporting serialize/deserialize using JSON schema to support loading in/from other languages

**Comment by sourabhdalvi** [comment]
> > * [x]   Add Auto-generated structs
> > * [x]   Add Serialization
> > * [ ]   Add De-serialization
> > * [ ]   Add Test
> 
> We discussed supporting serialize/deserialize using JSON schema to support loading in/from other languages

I have no problem with that, I or Rodrigo didn't remember that bit so we went ahead with this for now. Plus this should allow us to start working on the simulation side.

File: src/descriptors/portfolio_structs.json (hunk 1 above)
Comment (sourabhdalvi) [reply src/descriptors/portfolio_structs.json:1]: Eventually we will need to build more detailed CEM models for GMLC and this should not complicate it too much.

**Review by rodrigomha — APPROVED** [review]

**Pull request merged**